rfcs/0029-allow-script-multiple-matches-on-identical-code/0029-allow-script-multiple-matches-on-identical-code.md

```diff
@@ -1 +1,93 @@
-In review, see <https://github.com/nervosnetwork/rfcs/pull/222>
+---
+Number: "0029"
+Category: Consensus (Hard Fork)
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-02-03
+---
+
+# Allow Multiple Cell Dep Matches When There Is No Ambiguity
+
+## Abstract
+
+This document proposes a transaction verification consensus change to allow multiple cell dep matches on type script hash when all the matches are resolved to the same script code.
+
+## Motivation
+
+Currently, CKB does not allow multiple matches when resolving the script by type script hash in dep cells.
+
+However, in many cases, multiple matches do not introduce ambiguity.
+
+There are two different ways to reference the script, via data hash or type hash.
+
+* When the script is referenced by data hash, all the matches are resolved to the identical code.
+* When the script is referenced by type hash, it is possible that the matches can be resolved to different code binaries. The consensus rule can only reject the transaction when this happens.
+
+## Specification
+
+When the transaction verifier resolves script code in dep cell via data hash, multiple matches are allowed. This is the same as before.
```

Comment:
In the above section, we were talking about `referencing a script`, but here, it's `transaction verifier resolves script code`. There might be the benefit of unifying to a single set of terms, so as not to confuse people. We know they are the same thing, but someone else might need a confirmation.